main.tf

```diff
@@ -8,6 +8,8 @@ locals {
   validate_auth_policy = var.kms_encryption_enabled && var.skip_iam_authorization_policy == false && var.existing_kms_instance_guid == null ? tobool("When var.skip_iam_authorization_policy is set to false, and var.kms_encryption_enabled to true, a value must be passed for var.existing_kms_instance_guid in order to create the auth policy.") : true
   # tflint-ignore: terraform_unused_declarations
   validate_backup_key = var.backup_encryption_key_crn != null && var.use_default_backup_encryption_key == true ? tobool("When passing a value for 'backup_encryption_key_crn' you cannot set 'use_default_backup_encryption_key' to 'true'") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_plan = var.enable_elser_model && var.plan == "enterprise" ? tobool("When enabling elser configuration , var.plan should be 'platinum'") : true
```

Comment:
Let's check if the plan is not `platinum`, since currently the platinum is only available option for elser model
```
validate_plan = var.enable_elser_model && var.plan != "platinum"
```
error message:
`When var.enable_elser_model is set to true, a value for var.plan must be 'platinum' in order to enable ELSER model.`

Reply:
addressed